src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs

```diff
@@ -169,58 +169,43 @@ static ProjectCodeRefactoringProvider.ExtensionInfo GetExtensionInfo(ExportCodeR
             CodeActionOptionsProvider options,
             CancellationToken cancellationToken)
         {
-            cancellationToken.ThrowIfCancellationRequested();
-            if (extensionManager.IsDisabled(provider))
-            {
-                return null;
-            }
-
-            try
-            {
-                using var _ = ArrayBuilder<(CodeAction action, TextSpan? applicableToSpan)>.GetInstance(out var actions);
-                var context = new CodeRefactoringContext(textDocument, state,
-
-                    // TODO: Can we share code between similar lambdas that we pass to this API in BatchFixAllProvider.cs, CodeFixService.cs and CodeRefactoringService.cs?
-                    (action, applicableToSpan) =>
-                    {
-                        // Serialize access for thread safety - we don't know what thread the refactoring provider will call this delegate from.
-                        lock (actions)
-                        {
-                            // Add the Refactoring Provider Name to the parent CodeAction's CustomTags.
-                            // Always add a name even in cases of 3rd party refactorings that do not export
-                            // name metadata.
-                            action.AddCustomTagAndTelemetryInfo(providerMetadata, provider);
-
-                            actions.Add((action, applicableToSpan));
-                        }
-                    },
-                    options,
-                    cancellationToken);
-
-                var task = provider.ComputeRefactoringsAsync(context) ?? Task.CompletedTask;
-                await task.ConfigureAwait(false);
-
-                if (actions.Count == 0)
+            return extensionManager.PerformFunctionAsync(
```

Comment:
moved all explicit try/catch code with extension manager to use the existing helper functions that already do the right thing.